manifests/params.pp

```diff
@@ -20,42 +20,27 @@
   $package_ensure   = 'present'
   $package_name     = 'hiera'
   $hierarchy        = []
-  if str2bool($facts['is_pe']) {
-    $hiera_yaml     = '/etc/puppetlabs/puppet/hiera.yaml'
-    $datadir        = '/etc/puppetlabs/puppet/hieradata'
-    $owner          = 'pe-puppet'
-    $group          = 'pe-puppet'
-    $cmdpath        = ['/opt/puppet/bin', '/usr/bin', '/usr/local/bin']
-    $manage_package = false
-
-    if $::pe_version and versioncmp($::pe_version, '3.7.0') >= 0 {
-      $provider       = 'pe_puppetserver_gem'
-      $master_service = 'pe-puppetserver'
-    } else {
-      $provider       = 'pe_gem'
-      $master_service = 'pe-httpd'
-    }
-  } else {
-    # Configure for AIO packaging.
-    if getvar('::pe_server_version') {
-      $master_service = 'pe-puppetserver'
-      $provider       = 'puppetserver_gem'
-    } else {
-      # It would probably be better to assume this is puppetserver, but that
-      # would be a backwards-incompatible change.
-      $master_service = 'puppetmaster'
-      $provider       = 'puppet_gem'
-    }
-    $cmdpath        = ['/opt/puppetlabs/puppet/bin', '/usr/bin', '/usr/local/bin']
-    $datadir        = '/etc/puppetlabs/code/environments/%{::environment}/hieradata'
-    $manage_package = false
-    if getvar('::pe_server_version') {
-      $owner = 'pe-puppet'
-      $group = 'pe-puppet'
-    } else {
-      $owner = 'puppet'
-      $group = 'puppet'
-    }
-    $hiera_yaml = "${confdir}/hiera.yaml"
+  # Configure for AIO packaging.
+  if getvar('::pe_server_version') {
```

Comment:
This is a good opportunity to stop using `getvar`.  It's a leftover from puppet 3.x days.

Reply:
What is recommended instead?

Reply:
This codes predates the `fact` hash, so these days, I'd use that.
```puppet
if $facts['pe_server_version'] {
```

It really was an awkward time.  strict_variables was introduced but functions like `defined` were broken in certain versions of puppet etc.